lessons/cosmosdb.md

```diff
@@ -32,8 +43,339 @@ In this exercise we'll look into Cosmos DB Emulator to see how you can interact
+  }
+}
+
+```
+
+Create the Cosmos DB database locally
+
+We need to have the Cosmos DB emulator for running everything in our local environment. 
+For using the Cosmos DB emulator in OS X follow the official guide from the Microsoft Docs at this [link](https://docs.microsoft.com/en-us/azure/cosmos-db/local-emulator?tabs=cli%2Cssl-netstd21#run-on-linux-macos) 
+
+In summary, you will have to create a Windows Virtual Machine hosted in Parallels or Virtual Box. Then establish the connection between the host and the guest machines and finally set up the certificate to use the HTTPS connection in OS X. If you run into any issues with this setup you can take a look at this [github issue](https://github.com/Azure/Azure-Functions/issues/1797) for troubleshooting or you can opt for creating an actual Cosmos DB in Azure instead of using the emulator.
+
+Open your Cosmos DB Emulator and select the `New Database` button at the toolbar. We will name it `Players`. The next step is adding a new container. Keep in mind that the actual data of a Cosmos DB is stored in containers. Let's add a new one named `players`. Here is where the data from the queue will be saved.
+
+Take a look at the below image for reference.
+
+![Azure Cosmos DB Emulator](../img/lessons/cosmos/create-a-new-database-cosmos.png) 
+
+When you add the new container you have to define the container key. This is a very important setting, since this will be used to scale your Cosmos DB. The container key should be picked based on the distribution of the data. Here is an extract from the Microsoft docs:
+
+>"If you choose a partition key that evenly distributes throughput consumption across logical partitions, you will ensure that throughput consumption across physical partitions is balanced."
+
+
+For this exercise, our Partition Key will be `region`, and also we will add a unique key using the `playerId` field.
+
+![Container and partition key](../img/lessons/cosmos/add-container-and-partition.png) 
+
+
+> 🔎 **Observation** - If you are interested in learning more about how the partition key impacts the performance of your application check out [this](https://docs.microsoft.com/en-us/azure/cosmos-db/partitioning-overview) page from the official docs.
+
+Add the connection string to your Azure Function.
+
+Now that we have the Cosmos DB created locally, let's set up the Azure Function to use the local connection string.
+
+Get the connection string from the local emulator:
+
+![Connection string from cosmos db emulator](../img/lessons/cosmos/primary-connection-string-cosmos.png) 
+
+Since we are running the local emulator in a hosted virtual machine we have to change the IP of localhost to the IP of the virtual machine. After getting the IP from the VM, remove localhost and use the actual IP address. Finally add it to the local.settings.json file.
+
+```json
+{
+  "IsEncrypted": false,
+  "Values": {
+    "AzureWebJobsStorage": "UseDevelopmentStorage=false",

```

Comment:
So setting this to `false` was essential for the app to work right? If so, that would be good to add an obversion for this one.

Reply:
Yes, it was required to have it in false, good point about adding the observation. I will do that.